utils/python/CIME/XML/env_mach_pes.py

```diff
@@ -95,6 +95,8 @@ def get_total_tasks(self, comp_classes):
             pstrid = self.get_value("PSTRID_%s"%comp)
             tt = rootpe + (ntasks - 1) * pstrid + 1
             total_tasks = max(tt, total_tasks)
+        if total_tasks < self.get_value("PES_PER_NODE"): # all PEs fit in 1 node
```

Comment:
I'm somewhat unclear on the difference between MAX_TASKS_PER_NODE and PES_PER_NODE. I think MAX_TASKS_PER_NODE is actually max PEs per node. PE is either an MPI task (rank) or an OMP thread. And PES_PER_NODE is a user-editable value: i.e. a case can be run with different number of PEs per node.

In the above scenario, if a user changes number of ranks or threads for any of the components so that total_tasks becomes greater than PES_PER_NODE, then the user knows what they are doing and should also change PES_PER_NODE. Also, there are very few cases where total_tasks is less than PES_PER_NODE: most cases require more than 1 node to run on.